website/api/controllers/entrance/signup.js

```diff
@@ -78,13 +83,86 @@ the account verification message.)`,
+    let fleetSandboxDemoKey = await sails.helpers.strings.uuid();
+
+    // Send a POST request to the cloud provisioner API
+    let cloudProvisionerResponseData = await sails.helpers.http.post(
+      'https://sandbox.fleetdm.com/new',
+      { // Request body
+        'name': firstName + ' ' + lastName,
+        'email': newEmailAddress,
+        'password': fleetSandboxDemoKey, //« this provisioner API was originally designed to accept passwords, but rather than specifying the real plaintext password, since users always access Fleet Sandbox from their fleetdm.com account anyway, this generated demo key is used instead to avoid any confusion
+        'sandbox_expiration': new Date(fleetSandboxExpiresAt).toISOString(), // sending expiration_timestamp as an ISO string.
+      },
+      { // Request headers
+        'Authorization':sails.config.custom.cloudProvisionerSecret
+      }
+    )
+    .timeout(5000)
+    .intercept(['requestFailed', 'non200Response'], (err)=>{
+      // If we recieved a non-200 response from the cloud provisioner API, we'll throw a 500 error.
+      return new Error('When attempting to provision a new user who just signed up ('+emailAddress+'), the cloud provisioner gave a non 200 response. The incomplete user record has not been saved in the database, and the user will be asked to try signing up again. Raw response received from provisioner: '+err.stack);
+    });
+
+    if(!cloudProvisionerResponseData.URL) {
+      // If we didn't receive a URL in the response from the cloud provisioner API, we'll throwing an error before we save the new user record and the user will need to try to sign up again.
+      throw new Error(
+        `When provisioning a Fleet Sandbox instance for a new user who just signed up (${emailAddress}), the response data from the cloud provisioner API was malformed. It did not contain a valid Fleet Sandbox instance URL in its expected "URL" property.
+        The incomplete user record has not been saved in the database, and the user will be asked to try signing up again.
+        Here is the malformed response data (parsed response body) from the cloud provisioner API: ${cloudProvisionerResponseData}`
+      );
+    }
+
+    // If "Try Fleet Sandbox" was provided as the signupReason, we'll send a request to Zapier to add this user to our CRM and make sure their Sandbox instance is live before we continue.
+    if(signupReason === 'Try Fleet Sandbox') {
+      // Send a POST request to Zapier
+      await sails.helpers.http.post(
+        'https://hooks.zapier.com/hooks/catch/3627242/bqsf4rj/',
+        { 'emailAddress': newEmailAddress},
+        {'x-webhook-secret': sails.config.custom.zapierSandboxWebhookSecret}
+      )
+      .timeout(5000)
+      .tolerate(['non200Response', 'requestFailed'], (err)=>{
+        sails.log.warn(`When a new user signed up for Fleet Sandbox, a lead/contact could not be verified in the CRM for this email address: ${newEmailAddress}. Raw error: ${err}`);
```

Comment:
> One thing to note is that the Zapier webhook will always return a 200 response, the check for the x-webhook-secret header happens after Zapier receives the request (as a separate step in the zap).

👍 
<img width="784" alt="image" src="https://user-images.githubusercontent.com/618009/182756658-dd9dec3a-6d02-4f8f-94f4-ae1475b98f24.png">

Makes sense.  Probably worth leaving a link to [your related comment](https://github.com/fleetdm/fleet/pull/6380#issuecomment-1204395762) as a code comment here for when we're puzzling over this in the future.